app/src/main/java/com/kickstarter/libs/utils/AnalyticEventsUtils.kt

```diff
@@ -0,0 +1,226 @@
+package com.kickstarter.libs.utils
+
+import com.kickstarter.libs.RefTag
+import com.kickstarter.libs.utils.RewardUtils.isItemized
+import com.kickstarter.libs.utils.RewardUtils.isShippable
+import com.kickstarter.libs.utils.RewardUtils.isTimeLimitedEnd
+import com.kickstarter.libs.utils.extensions.totalAmount
+import com.kickstarter.models.*
+import com.kickstarter.services.DiscoveryParams
+import com.kickstarter.ui.data.CheckoutData
+import com.kickstarter.ui.data.PledgeData
+import java.util.*
+import kotlin.math.ceil
+import kotlin.math.roundToInt
+
+object AnalyticEventsUtils {
+
+    @JvmOverloads
+    fun checkoutProperties(checkoutData: CheckoutData, pledgeData: PledgeData, prefix: String = "checkout_"): Map<String, Any> {
+        val project = pledgeData.projectData().project()
+        val properties = HashMap<String, Any>().apply {
+            put("amount", checkoutData.amount())
+            checkoutData.id()?.let { put("id", it) }
+            put("payment_type", checkoutData.paymentType().rawValue())
+            put("amount_total_usd", checkoutData.totalAmount() * project.staticUsdRate())
+            put("shipping_amount", checkoutData.shippingAmount())
+            if (checkoutData.bonusAmount() != null) {
```

Comment:
Do we need this null check since we are checking for null inside of the condition using a `let` function?